testnet/stacks-node/src/tests/neon_integrations.rs

```diff
@@ -0,0 +1,212 @@
+    conf.burnchain.peer_host = "127.0.0.1".into();
+    conf.burnchain.local_mining_public_key = Some(keychain.generate_op_signer()
+        .get_public_key().to_hex());
+    conf.burnchain.commit_anchor_block_within = 0;
+
+    let miner_account = keychain.origin_address().unwrap();
+
+    (conf, miner_account)
+}
+
+#[test]
+#[ignore]
+fn bitcoind_integration_test() {
+    if env::var("BITCOIND_TEST") != Ok("1".into()) {
+        return
+    }
+
+    // used to specify how long to wait in between blocks.
+    //   we could _probably_ add a hook to the neon node that
+    //   would remove some of the need for this
+    let gap_ms = 5_000;
+
+    let (conf, miner_account) = neon_integration_test_conf();
+
+    let mut btcd_controller = BitcoinCoreController::new(conf.clone());
+    btcd_controller.start_bitcoind().map_err(|_e| ()).expect("Failed starting bitcoind");
+
+    let mut btc_regtest_controller = BitcoinRegtestController::new(conf.clone());
+    let http_origin = format!("http://{}", &conf.node.rpc_bind);
+
+    btc_regtest_controller.bootstrap_chain(201);
+
+    eprintln!("Chain bootstrapped...");
+
+    let mut run_loop = neon::RunLoop::new(conf);
+    let client = reqwest::blocking::Client::new();
+
+    thread::spawn(move || {
+        run_loop.start(0)
+    });
+
+    // give the run loop some time to start up!
+    thread::sleep(std::time::Duration::from_millis(gap_ms));
+
+    // first block wakes up the run loop
+    btc_regtest_controller.build_next_block(1);
+    eprintln!("== REGTEST BLOCK MINED == ");
+    // give the run loop some time to figure things out!
+    thread::sleep(std::time::Duration::from_millis(gap_ms));
+
+    // first block will hold our VRF registration
+    btc_regtest_controller.build_next_block(1);
+    eprintln!("== REGTEST BLOCK MINED == ");
+    // give the run loop some time to figure things out!
+    thread::sleep(std::time::Duration::from_millis(gap_ms));
+
+    // second block will be the first mined Stacks block
+    btc_regtest_controller.build_next_block(1);
+    eprintln!("== REGTEST BLOCK MINED == ");
+    thread::sleep(std::time::Duration::from_millis(gap_ms));
+
+    // let's query the miner's account nonce:
+
+    eprintln!("Miner account: {}", miner_account);
+
+    let path = format!("{}/v2/accounts/{}?proof=0",
+                       &http_origin, &miner_account);
+    eprintln!("Test: GET {}", path);
+    let res = client.get(&path).send().unwrap().json::<AccountEntryResponse>().unwrap();
+    eprintln!("Response: {:#?}", res);
+    assert_eq!(u128::from_str_radix(&res.balance[2..], 16).unwrap(), 0);
+    assert_eq!(res.nonce, 1);
+}
+
+#[test]
+#[ignore]
+fn microblock_integration_test() {
+    if env::var("BITCOIND_TEST") != Ok("1".into()) {
+        return
+    }
+
+    // used to specify how long to wait in between blocks.
+    //   we could _probably_ add a hook to the neon node that
+    //   would remove some of the need for this
+    let gap_ms = 5_000;
```

Comment:
How hard would this be?  I'm not comfortable with non-deterministic testing if it can be avoided.  Can we do something like poll the neon node's `/v2/info` endpoint to see how far it is in processing the burnchain?  Can we add an API endpoint for getting the current Stacks chain tip and its height, so we can poll on that too?

Reply:
Depends on how it is implemented. Unfortunately, just polling `/v2/info/` wouldn't work, because the thing we want to wait on is the node finishing all of the processing for a new burnchain block (e.g., if it can mine a block, that it has done so and broadcasted it's transaction, or if it's just trying to register a VRF key, that it has done that). I'll add a (test-only) Mutex guarded field to the neon node for 'blocks_processed', which could be monitored rather than relying on the gap.

Reply:
Right -- we need a way to poll on the Stacks chain tip.  This could be part of `/v2/info`, but it currently isn't now.

Reply:
I don't think even that would be enough here -- we're trying to wait for the node to have finished broadcasting it's block commit, which wouldn't be reflected in the chain state until the next sortition.

Reply:
Added in 0dfa4cc1f

Reply:
Thanks! Looks much better